backend/src/service/reload.service.ts

```diff
@@ -110,6 +112,23 @@ export default class ReloadService {
     } else {
       this.repositoryService.southConnectorRepository.stopSouthConnector(southId);
     }
+
+    if (previousSettings.history.maxInstantPerItem != command.history.maxInstantPerItem) {
+      const maxInstants = this.repositoryService.southCacheRepository.getLatestMaxInstants(southId);
+      const latestInstant = maxInstants?.values().next().value as Instant;
+      const southItems = this.repositoryService.southItemRepository.getSouthItems(southId);
+      this.repositoryService.southCacheRepository.deleteAllCacheScanModes(southId);
+      if (!maxInstants) return;
+
+      for (const item of southItems) {
+        this.repositoryService.southCacheRepository.createOrUpdateCacheScanMode({
+          southId,
+          itemId: command.history.maxInstantPerItem ? item.id : 'all', // When max instant per item is disabled, we use 'all' as item_id
+          scanModeId: item.scanModeId,
+          maxInstant: maxInstants.get(item.scanModeId) || latestInstant
+        });
+      }
+    }
```

Comment:
This should be called before the if(command.enabled) line because if south connector restart, it will take the previous maxInstant